src/qibo/models/encodings.py

```diff
@@ -435,10 +435,20 @@ def _generate_rbs_angles(data, nqubits: int, architecture: str):
         list: list of phases for RBS gates.
     """
     if architecture == "diagonal":
-        phases = [
-            math.atan2(np.linalg.norm(data[k + 1 :]), data[k])
-            for k in range(len(data) - 2)
-        ]
+        if data.__class__.__name__ == "Tensor":
+            import torch  # pylint: disable=C0415
```

Comment:
this pattern here is repeated several times, maybe at this point it could be useful to define a helper function to set the engine, something like:

``` python
def _check_torch(array: "ndarray") -> "engine":
    if array.__class__.__name__ == "Tensor":
        import torch

        return torch
    else:
        return np
```

and then always call methods from the engine

``` python
engine = _check_torch(array)
engine.method_needed()
```